### PR TITLE
Add configuration to support broker SSL

### PR DESCRIPTION
**Pull request by eapache**
Possibly implements #154, if my assumptions about the implementation are
correct.

@joestein @wvanbergen

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -12,6 +15,11 @@ type Config struct {
 		DialTimeout  time.Duration // How long to wait for the initial connection to succeed before timing out and returning an error (default 30s).
 		ReadTimeout  time.Duration // How long to wait for a response before timing out and returning an error (default 30s).
 		WriteTimeout time.Duration // How long to wait for a transmit to succeed before timing out and returning an error (default 30s).
+
+		TLS struct {
+			Enable bool        // Whether or not to use TLS when connecting to the broker (defaults to false).
+			Config *tls.Config // The TLS configuration to use for secure connections if specified by UseTLS (defaults to nil).
```
Comment (wvanbergen) [inline config.go]: godoc: `UseTLS` is probably an old name, and should be `Enable`?

Comment (eapache) [inline config.go]: Good catch, this was left over from a rebase before unified config.

**Comment by eapache** [comment]
@wvanbergen @uovobw Rebased this branch correctly and added a note about compat guarantees. If this looks good to everyone I'll merge it.

**Comment by wvanbergen** [comment]
:tada: Does it use a default config if you leave the config set to `nil`?

I'd like to wait for a confirmation that this works as expected for @uovobw before merging.

**Comment by eapache** [comment]
>  DialWithDialer interprets a nil configuration as equivalent to the zero configuration; see the documentation of Config for the defaults.

**Comment by uovobw** [comment]
@wvanbergen thanks for the trust but i no longer have the infrastructure in place to test this completely (since broker-originating connections do not support ssl) and currently do not have time to setup a local broker and test against this. if this merge is not urgent i think i can on this in a couple of days. thanks for the feedback guys

**Comment by wvanbergen** [comment]
@uovobw Given that you are the only person we know of that needs this feature, you can take your time :)

**Pull request closed without merging**